scanner.go

```diff
@@ -88,11 +92,19 @@ func (s *scanner) peek() (*pb.Result, error) {
 
 	if s.closed {
 		// done scanning
+		if s.rpc.ScanStatsHandler() != nil {
+			updateScanStats(s.rpc.ScanStats(), s.rpc, s, nil, time.Since(s.start))
+			s.rpc.ScanStatsHandler()(s.rpc.ScanStats())
+		}
```

Comment:
I was imagining that we would emit a ScanStats for every single scan including the partial scans. ie. Emit one of these inside `request()` for every call to `SendRPC`. Going a step further I would like to be able to emit a ScanStats for every failure and retry inside `SendRPC`.

The reason to do this is that the sub-scans may end up going to different regions or regionservers, so to track that properly we need to emit a scanstats for each of those requests. Also, I wanted to capture the cases where we are hitting an overloaded region server, getting CallQueueTooBig exceptions and retrying.